runtime/src/chain_extension/tests.rs

```diff
@@ -0,0 +1,367 @@
+
+			assert_ok!(
+				Fragments::mint(
+					Origin::signed(ALICE),
+					definition_hash,
+					pallet_fragments::FragmentBuyOptions::Quantity(1),
+					None,
+				)
+			);
+
+			let contract_result = Contracts::bare_call(
+				ALICE,
+				contract_address, // Address of the contract to call.
+				0, // The balance to transfer from the origin to dest.
+				GAS_LIMIT, // The gas limit enforced when executing the constructor.
+				None, // The maximum amount of balance that can be charged from the caller to pay for the storage consumed.
+				vec![&blake2_256(b"get_instance")[0..4], &(definition_hash, 1u64, 1u64).encode()[..]].concat(), // The input data to pass to the contract.
+				false // `debug` should only ever be set to true when executing as an RPC because it adds allocations and could be abused to drive the runtime into an OOM panic.
+			);
+
+			assert_eq!(contract_result.result.as_ref().unwrap().flags.bits(), 0);
+			assert_eq!(
+				contract_result.result.unwrap().data.0,
+				Some(pallet_fragments::Fragments::<Test>::get(&(definition_hash, 1, 1)).unwrap()).encode()
+			);
+
+		});
+	}
+
+	#[test]
+	pub fn get_instance_ids_should_work() {
+		new_test_ext().execute_with(|| {
+
+			let code_hash = upload_dummy_contract(ALICE);
+			let contract_address = Contracts::contract_address(&ALICE, &code_hash, &[]);
+
+			let proto_data = b"Je suis Data".to_vec();
+			assert_ok!(upload(ALICE, &proto_data));
+			let proto_hash = blake2_256(&proto_data);
+
+			let definition_name = b"Je suis un Nom".to_vec();
+			assert_ok!(create(ALICE, &proto_hash, &definition_name));
+			let definition_hash = blake2_128(&[&proto_hash[..], &definition_name.encode(), &pallet_fragments::Currency::<<Test as pallet_assets::Config>::AssetId>::Native.encode()].concat());
+
+			assert_ok!(
+				Fragments::mint(
+					Origin::signed(ALICE),
+					definition_hash,
+					pallet_fragments::FragmentBuyOptions::Quantity(10),
+					None,
+				)
+			);
+
+			let contract_result = Contracts::bare_call(
+				ALICE,
+				contract_address, // Address of the contract to call.
+				0, // The balance to transfer from the origin to dest.
+				GAS_LIMIT, // The gas limit enforced when executing the constructor.
+				None, // The maximum amount of balance that can be charged from the caller to pay for the storage consumed.
+				vec![&blake2_256(b"get_instance_ids")[0..4], &(definition_hash, ALICE).encode()[..]].concat(), // The input data to pass to the contract.
+				false // `debug` should only ever be set to true when executing as an RPC because it adds allocations and could be abused to drive the runtime into an OOM panic.
+			);
+
+			assert_eq!(contract_result.result.as_ref().unwrap().flags.bits(), 0);
+			assert_eq!(
+				contract_result.result.unwrap().data.0,
+				pallet_fragments::Inventory::<Test>::get(ALICE, definition_hash).unwrap().encode()
+			);
+
+		});
+	}
+
+	#[test]
+	pub fn give_instance_should_work() {
```

Comment:
add a should not work if not authorized and such

Reply:
> add a should not work if not authorized and such

added